plugin-hrm-form/src/services/InsightsService.ts

```
@@ -209,107 +209,129 @@ type InsightsCaseForm = {
+       * customization framework.
+       */
+      const thePerp = caseForm.info.perpetrators[0];
+      const untypedPerp: any = {
+        ...thePerp,
+        ...thePerp.perpetrator,
+      };
+      delete untypedPerp.perpetrator;
+      delete untypedPerp.name;
+      delete untypedPerp.location;
+      perpetrator = {
+        ...untypedPerp,
+      };
+    }
+    if (caseForm.info?.incidents && caseForm.info.incidents.length > 0) {
+      const theIncident = caseForm.info.incidents[0];
+      const untypedIncident: any = {
+        ...theIncident,
+        ...theIncident.incident,
+      };
+      delete untypedIncident.incident;
+      incident = {
+        ...untypedIncident,
+      };
+    }
+    if (caseForm.info?.households && caseForm.info.households.length > 0) {
+      const theHousehold = caseForm.info.households[0];
+      const untypedHousehold: any = {
+        ...theHousehold,
+        ...theHousehold.household,
+      };
+      delete untypedHousehold.household;
+      household = {
+        ...untypedHousehold,
+      };
+    }
+    if (caseForm.info?.referrals && caseForm.info.referrals.length > 0) {
+      referral = {
+        ...caseForm.info.referrals[0],
+        date: caseForm.info.referrals[0].date.toString(),
+      };
+    }
+
+    return {
+      topLevel,
+      perpetrator,
+      incident,
+      household,
+      referral,
     };
-  }
-  if (caseForm.info?.referrals && caseForm.info.referrals.length > 0) {
-    referral = {
-      ...caseForm.info.referrals[0],
-      date: caseForm.info.referrals[0].date.toString(),
+  } catch (error) {
+    const errorObject = {
+      message: error,
+      contactsDetails: caseForm.connectedContacts.map(({ channel, taskId }) => ({ channelType: channel, taskId })),
+      accountSid: caseForm.accountSid,
+      twilioWorkerId: caseForm.twilioWorkerId,
     };
+    console.error(errorObject);
+    throw new Error(error);
```

Can you rethrow the original error rather than a new one so we retain the call stack? I'm not sure it's actually valid to construct an error with another error, TS is only allowing it because catch parameters come through as 'any' types in TS

---

@stephenhand - The original error is what was thrown last in the  `throw new Error(error)` block. The `console.error` block contains some additional info including the original error as a message object